extensions/WSLExtension/Constants.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace WSLExtension;
+
+public static class Constants
+{
+#if CANARY_BUILD
+    public const string ExtensionIcon = "ms-resource://Microsoft.Windows.DevHome.Canary/Files/WslAssets/wslLinux.png";
+#elif STABLE_BUILD
+    public const string ExtensionIcon = "ms-resource://Microsoft.Windows.DevHome/Files/WslAssets/wslLinux.png";
+#else
+    public const string ExtensionIcon = "ms-resource://Microsoft.Windows.DevHome.Dev/Files/WslAssets/wslLinux.png";
+#endif
+
+    // Common unlocalized strings used for WSL extension
+    public const string WslProviderDisplayName = "Microsoft WSL";
+    public const string WslProviderId = "Microsoft.WSL";
+    public const string WindowsTerminalShimExe = "wt.exe";
+    public const string WindowsTerminalPackageFamilyName = "Microsoft.WindowsTerminal_8wekyb3d8bbwe";
+    public const string WslExe = "wsl.exe";
+    public const string WslTemplateSubfolderName = "WslTemplates";
+
+    public const string DefaultWslLogoPath = @"ms-appx:///WslAssets/wslLinux.png";
+    public const string WslLogoPathFormat = @"ms-appx:///WslAssets/{0}";
+    public const string KnownDistributionsLocalYamlLocation = @"ms-appx:///DistributionDefinitions/DistributionDefinition.yaml";
+    public const string KnownDistributionsWebJsonLocation = @"https://aka.ms/wsldistributionsjson";
+
+    // Wsl registry location for registered distributions.
+    public const string WslRegistryLocation = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss";
+
+    // Wsl registry data names within a distribution location.
+    public const string PackageFamilyRegistryName = "PackageFamilyName";
+    public const string DistributionRegistryName = "DistributionName";
+    public const string DefaultDistributionRegistryName = "DefaultDistribution";
+    public const string WslVersion = "Version";
+    public const string WslState = "State";
+    public const int WslVersion1 = 1;
+    public const int WslVersion2 = 2;
+    public const int WslExeExitSuccess = 0;
+
+    // Launch terminal with specific profile and log the user into their home directory in the login shell
+    // Note: this opens a new terminal window in the UI
+    public static string LaunchDistributionInTerminalWithProfile { get; } = "--profile {0} -- wsl --shell-type login --cd ~ --distribution {1}";

```

Comment:
NIT for this and the other strings.  Explain what the placeholders are for.  This way calling code does not need to look at the string.

Another option is to remove the properties all together and change these into methods.  Then, the code can describe the format with parameters.

Reply:
I don't think it would make sense to make them into methods, because in the future when the WSL team provides us with apis then we'd need to remove/change those methods. Today we're using wsl.exe under the hood, but I've made it in a way where, when we get new apis, only the body of the methods that use these strings need to change, not the method themselves. 

Ultimately if you typed in `wsl --help` in your command line window you'll see what these commands are used for but I can add it to the comments if you feel strongly about it.